app/code/Bulbulatory/Recomendations/Block/Recommendation/Index.php

```diff
@@ -20,7 +36,52 @@ public function getFormAction()
 
     protected function _prepareLayout()
     {
-        return parent::_prepareLayout();
+        parent::_prepareLayout();
+
+        $this->pageConfig->getTitle()->set(__('Recommendation'));
+
+        if ($this->getRecommendationCollection()) {
+            $pager = $this->getLayout()->createBlock('Magento\Theme\Block\Html\Pager','custom.recommendation.pager')
+                                       ->setAvailableLimit([5 => 5, 10 => 10, 15 => 15, 20 => 20])
+                                       ->setShowPerPage(true)
+                                       ->setCollection($this->getRecommendationCollection());
+            $this->setChild('pager', $pager);
+            $this->getRecommendationCollection()->load();
+        }
+        return $this;
+    }
+
+    public function getPagerHtml()
+    {
+        return $this->getChildHtml('pager');
     }
- 
+
+    public function getRecommendationCollection()
+    {
+        $page = ($this->getRequest()->getParam('p')) ? $this->getRequest()->getParam('p') : 1;
+        $pageSize = ($this->getRequest()->getParam('limit')) ? $this->getRequest()->getParam('limit') : 5;
+
+        $customer = $this->customer;
+        $recommenderId = $customer->getId();
+
+        $collection = $this->recommendationCollection->create();
+        $collection->setPageSize($pageSize);
+        $collection->setCurPage($page);
+        $collection->addFieldToFilter('recommender_id', $recommenderId);
+        return $collection;
+    } 
+
+    public function getConfirmRecommendationCollection()
+    {
+        $collection = $this->getRecommendationCollection();
+        $collection->addFieldToFilter('status', 1);
+        return $collection->getSize();
+    } 
+
+    public function calculateDiscount()
```

Comment:
Obliczamy tutaj zniżkę procentową, więc może lepiej byłoby nazwać funkcję `calculateDiscountPercent`, bo teraz jeśli ktoś nowy na to spojrzy, to może pomyśleć że to zniżka -X złotych.